Breaking change docs: `dotnet nuget why` output format changed in .NET 10.0.400 SDK (#53562)
dotnet/docs: docs/core/compatibility/sdk/10.0/dotnet-nuget-why-output-format.md

Previously, `dotnet nuget why` displayed each package in the dependency graph using only the resolved version, in the format `<PackageName> (v<ResolvedVersion>)`.

For example:

```output
System.Text.Json (v9.0.0-rc.2.24427.10)
```

## New behavior

Starting in .NET 10.0.400 SDK, `dotnet nuget why` displays each package using the resolved version and the minimum requested version constraint, in the format `<PackageName>@<ResolvedVersion> (>= <RequestedVersion>)`.

For example:

```output
System.Text.Json@10.0.1 (>= 9.0.0)
```

## Type of breaking change

This change is a [behavioral change](../../categories.md#behavioral-change).

## Reason for change

To help you investigate why a transitive package resolves to a particular version, the output now shows both the version that each dependent package requested and the version that NuGet actually resolved. This information makes it easier to understand version conflicts and dependency requirements.

## Recommended action

If you use a regular expression or other string parsing to process the output of `dotnet nuget why`, update your parsing logic to handle the new format `<PackageName>@<ResolvedVersion> (>= <RequestedVersion>)`.

> If some team members or CI agents might run older versions of the .NET SDK, update the parsing logic to support both formats:

Reviewer: ```suggestion
If your colleagues or CI agents run older versions of the .NET SDK, update the parsing logic to support both formats:
```
